pkg/validation/config.go

```diff
@@ -24,13 +24,19 @@ type Validator struct {
 
 // NewValidator creates an object that optimizes bulk validations.
 func NewValidator() Validator {
-	profiles := api.ClusterProfiles()
+	return Validator{
+		hasTrapCache: make(map[string]bool),
+	}
+}
+
+// NewCPValidator creates an object that  and optimizes bulk validations.
+func NewCPValidator(profiles api.ClusterProfilesList) Validator {
 	ret := Validator{
-		validClusterProfiles: make(clusterProfileSet, len(profiles)),
+		validClusterProfiles: make(clusterProfileSet, len(api.ClusterProfiles())), // TODO: replace with 'len(profiles)' once the config file exists
 		hasTrapCache:         make(map[string]bool),
 	}
-	for _, x := range profiles {
-		ret.validClusterProfiles[x] = struct{}{}
+	for _, p := range api.ClusterProfiles() { // TODO: replace with 'profiles' once the config file exists
+		ret.validClusterProfiles[p] = api.ClusterProfileDetails{} // TODO: initialize with owner info once the config file exists
 	}
```

Comment:
Why not keep the existing code (modulo changes to the value type of the `map`), since we're already requiring a profile list as a parameter?

Reply:
Yes that's what I plan to do, this is just an intermediate step so all the existing tests pass, while this is still a WIP